api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -206,6 +207,17 @@ message FilterChain {
   // name is provided, Envoy will allocate an internal UUID for the filter chain. If the filter
   // chain is to be dynamically updated or removed via FCDS a unique name must be provided.
   string name = 7;
+
+  // [#not-implemented-hide:] Whether the filter chain will be built on demand or not.
+  // The initialization of on-demand filter chains will be postponed until the first incoming traffic.
+  // On-demand filter chains will not block listener updates.
+  bool build_on_demand = 8;
+
+  // [#not-implemented-hide:] The timeout to wait for filter chain placeholder to complete rebuilding.
+  // If the timeout is reached, the filter chain rebuilding fails and sends callbacks to workers with failure signal
+  // so that workers will not retry connections that require this filter chain.
```

Comment:
Can you clarify what the end user behavior here is? I would remove "workers" etc. from the docs and make this end user behavior. What happens in the case of timeout? Are connections immediately closed? Etc.

Reply:
Thanks, I will added more description~